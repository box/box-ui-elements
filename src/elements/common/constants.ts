const DEFAULT_DATE_FORMAT = {
    year: 'numeric',
    month: 'short',
    day: 'numeric',
} as Intl.DateTimeFormatOptions;

export { DEFAULT_DATE_FORMAT };
