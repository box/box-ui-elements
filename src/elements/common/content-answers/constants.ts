/* ------------------ Allowable file extensions ------------------ */
export const CODE_FILE_EXTENSIONS = [
    'as',
    'as3',
    'asm',
    'bat',
    'c',
    'cc',
    'cmake',
    'cpp',
    'cs',
    'css',
    'cxx',
    'diff',
    'erb',
    'groovy',
    'h',
    'haml',
    'hh',
    'htm',
    'html',
    'java',
    'js',
    'json',
    'less',
    'm',
    'make',
    'md',
    'ml',
    'mm',
    'php',
    'pl',
    'plist',
    'properties',
    'py',
    'rb',
    'rst',
    'sass',
    'scala',
    'script',
    'scm',
    'sml',
    'sql',
    'sh',
    'vi',
    'vim',
    'webdoc',
    'xhtml',
    'yaml',
];

export const DOCUMENT_FILE_EXTENSIONS = [
    'csv',
    'doc',
    'docx',
    'dwg',
    'gdoc',
    'gsheet',
    'gslide',
    'gslides',
    'idml',
    'indd',
    'indt',
    'inx',
    'key',
    'numbers',
    'odp',
    'ods',
    'odt',
    'pages',
    'pdf',
    'ppt',
    'pptx',
    'rtf',
    'wpd',
    'xls',
    'xlsm',
    'xlsx',
];

export const TEXT_FILE_EXTENSIONS = [
    'as',
    'as3',
    'asm',
    'bat',
    'c',
    'cc',
    'cmake',
    'cpp',
    'cs',
    'css',
    'cxx',
    'diff',
    'erb',
    'groovy',
    'h',
    'haml',
    'hh',
    'java',
    'js',
    'json',
    'less',
    'log',
    'm',
    'make',
    'md',
    'ml',
    'mm',
    'php',
    'pl',
    'plist',
    'properties',
    'py',
    'rb',
    'rst',
    'sass',
    'scala',
    'script',
    'scm',
    'sml',
    'sql',
    'sh',
    'txt',
    'vi',
    'vim',
    'webdoc',
    'yaml',
];

export const SPREADSHEET_FILE_EXTENSIONS = ['xls', 'xlsx', 'xlsm', 'xlsb', 'csv', 'ods', 'gsheet'];

export const DOCUMENT_SUGGESTED_QUESTIONS = [
    {
        id: 'suggested-question-1',
        labelId: 'documentSuggestedQuestionLabel1',
        promptId: 'documentSuggestedQuestionPrompt1',
    },
    {
        id: 'suggested-question-2',
        labelId: 'documentSuggestedQuestionLabel2',
        promptId: 'documentSuggestedQuestionPrompt2',
    },
    {
        id: 'suggested-question-3',
        labelId: 'documentSuggestedQuestionLabel3',
        promptId: 'documentSuggestedQuestionPrompt3',
    },
    {
        id: 'suggested-question-4',
        labelId: 'documentSuggestedQuestionLabel4',
        promptId: 'documentSuggestedQuestionPrompt4',
    },
];
