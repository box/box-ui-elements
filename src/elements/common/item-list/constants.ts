const ITEM_LIST_COLUMNS = [
    {
        allowsSorting: true,
        id: 'name',
        isRowHeader: true,
        messageId: 'nameColumn',
    },
    {
        allowsSorting: true,
        id: 'date',
        messageId: 'dateColumn',
    },
    {
        allowsSorting: true,
        id: 'size',
        messageId: 'sizeColumn',
        width: 120,
    },
    {
        hideHeader: true,
        id: 'actions',
        messageId: 'actionsColumn',
        width: 100,
    },
];

const MEDIUM_ITEM_LIST_COLUMNS = [
    {
        allowsSorting: true,
        id: 'name',
        isRowHeader: true,
        messageId: 'nameColumn',
    },
    {
        allowsSorting: true,
        id: 'date',
        messageId: 'dateColumn',
    },
    {
        hideHeader: true,
        id: 'actions',
        messageId: 'actionsColumn',
        width: 100,
    },
];

const SMALL_ITEM_LIST_COLUMNS = [
    {
        id: 'details',
        isRowHeader: true,
        messageId: 'detailsColumn',
    },
    {
        hideHeader: true,
        id: 'actions',
        maxWidth: 10,
        messageId: 'actionsColumn',
        minWidth: 10,
    },
];

const ITEM_ICON_SIZE = 36;

export { ITEM_ICON_SIZE, ITEM_LIST_COLUMNS, MEDIUM_ITEM_LIST_COLUMNS, SMALL_ITEM_LIST_COLUMNS };
