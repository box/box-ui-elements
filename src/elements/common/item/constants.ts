const FILE_EXTENSIONS = {
    audio: ['aac', 'aif', 'aifc', 'aiff', 'amr', 'au', 'flac', 'm3u', 'm4a', 'mid', 'mp3', 'ra', 'wav', 'wma', 'wpl'],
    boxcanvas: ['boxcanvas'],
    boxnote: ['boxnote'],
    code: [
        'as',
        'as3',
        'asm',
        'aspx',
        'bat',
        'c',
        'cc',
        'cmake',
        'cpp',
        'cs',
        'css',
        'cxx',
        'db',
        'diff',
        'erb',
        'groovy',
        'h',
        'haml',
        'hh',
        'htm',
        'html',
        'java',
        'js',
        'json',
        'less',
        'm',
        'make',
        'md',
        'ml',
        'mm',
        'php',
        'pl',
        'plist',
        'properties',
        'py',
        'rb',
        'sass',
        'scala',
        'scm',
        'script',
        'sh',
        'sml',
        'sql',
        'wabba',
        'yaml',
    ],
    document: ['dot', 'dotx', 'msg', 'odt', 'rtf', 'wpd', 'xhtml', 'xml', 'xsd', 'xsl'],
    'docuworks-binder': ['xbd'],
    'docuworks-file': ['xdw'],
    dwg: ['dwg', 'dwgzip'],
    'excel-spreadsheet': ['xls', 'xlsb', 'xlsm', 'xlsx'],
    'google-docs': ['gdoc'],
    'google-sheets': ['gsheet'],
    'google-slides': ['gslide', 'gslides'],
    illustrator: ['ai'],
    image: ['bmp', 'gdraw', 'gif', 'jpeg', 'jpg', 'png', 'ps', 'svg', 'svs', 'tif', 'tiff', 'heic', 'heif'],
    indesign: ['idml', 'indd', 'indt', 'inx'],
    keynote: ['key'],
    numbers: ['numbers'],
    pages: ['pages'],
    pdf: ['pdf'],
    photoshop: ['psd'],
    'powerpoint-presentation': ['ppt', 'pptm', 'pptx'],
    presentation: ['odp', 'otp', 'pot', 'potx'],
    spreadsheet: ['csv', 'ods', 'tsv', 'xlt', 'xltx'],
    text: ['txt', 'vi', 'vim', 'webdoc'],
    threed: ['3ds', 'dae', 'fbx', 'obj', 'ply', 'stl'],
    vector: ['eps'],
    video: [
        '3g2',
        '3gp',
        'avi',
        'flv',
        'm2ts',
        'm2v',
        'm4v',
        'mkv',
        'mov',
        'mp4',
        'mpeg',
        'mpg',
        'mts',
        'ogg',
        'qt',
        'wmv',
    ],
    'word-document': ['doc', 'docm', 'docx'],
    zip: ['rar', 'tgz', 'zip'],
};

export { FILE_EXTENSIONS };
