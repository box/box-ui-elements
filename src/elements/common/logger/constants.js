// @flow
const EVENT_JS_READY = 'js_ready';
const EVENT_INITIALIZED = 'component_initialized';

export { EVENT_INITIALIZED, EVENT_JS_READY };
