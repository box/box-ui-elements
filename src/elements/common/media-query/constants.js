export const SIZE_SMALL = '(max-width: 374px)';
export const SIZE_MEDIUM = '(min-width: 375px) and (max-width: 767px)';
export const SIZE_LARGE = '(min-width: 768px) and (max-width: 1023px)';
export const SIZE_XLARGE = '(max-width: 1024px)';

export const HOVER = '(hover: hover)';
export const ANY_HOVER = '(any-hover: hover)';

export const POINTER_FINE = '(pointer: fine)';
export const POINTER_COARSE = '(pointer: coarse)';

export const ANY_POINTER_FINE = '(any-pointer: fine)';
export const ANY_POINTER_COARSE = '(any-pointer: coarse)';

export const POINTER_TYPE_FINE = 'fine';
export const POINTER_TYPE_COARSE = 'coarse';
export const POINTER_TYPE_NONE = 'none';

export const HOVER_TYPE_HOVER = 'hover';
export const HOVER_TYPE_NONE = 'none';

export const VIEW_SIZE = {
    SMALL: 'small',
    MEDIUM: 'medium',
    LARGE: 'large',
    XLARGE: 'x-large',
};

// enum ViewSize {
//     Small,
//         Medium,
//         Large,
//         XLarge
// }
