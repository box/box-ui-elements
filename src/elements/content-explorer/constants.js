import {
    FIELD_COLLECTIONS,
    FIELD_FILE_VERSION,
    FIELD_LOCK,
    FIELD_SHA1,
    FIELD_SHARED_LINK,
    FIELD_WATERMARK_INFO,
    FIELD_VERSION_NUMBER
} from '../../constants';
import { FOLDER_FIELDS_TO_FETCH } from '../../utils/fields';

// Fields needed for Content Explorer folder requests
const CONTENT_EXPLORER_FOLDER_FIELDS_TO_FETCH = [
    ...FOLDER_FIELDS_TO_FETCH,
    FIELD_FILE_VERSION,
    FIELD_SHA1,
    FIELD_SHARED_LINK,
    FIELD_WATERMARK_INFO,
    FIELD_COLLECTIONS,
    FIELD_LOCK,
    FIELD_VERSION_NUMBER
];

export default CONTENT_EXPLORER_FOLDER_FIELDS_TO_FETCH;
