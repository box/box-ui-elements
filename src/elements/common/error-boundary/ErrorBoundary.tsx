import * as React from 'react';
import noop from 'lodash/noop';
import DefaultError, { ErrorComponentProps } from './DefaultError';
import { ERROR_CODE_UNEXPECTED_EXCEPTION, IS_ERROR_DISPLAYED } from '../../../constants';
import type { ElementsXhrError, ElementsError } from '../../../common/types/api';
import type { ElementOrigin } from '../flowTypes';

export interface ErrorBoundaryProps {
    children: React.ReactElement;
    errorComponent: React.ComponentType<ErrorComponentProps>;
    errorOrigin: ElementOrigin;
    onError: (error: ElementsError) => void;
}

type State = {
    error?: Error;
};

class ErrorBoundary extends React.Component<ErrorBoundaryProps, State> {
    static defaultProps = {
        errorComponent: DefaultError,
        onError: noop,
    };

    state: State = {};

    componentDidCatch(error: Error, info: React.ErrorInfo): void {
        this.setState({ error }, () => {
            this.handleError(
                error,
                ERROR_CODE_UNEXPECTED_EXCEPTION,
                {
                    ...info,
                },
                this.props.errorOrigin,
            );
        });
    }

    /**
     * Formats the error and emits it to the top level onError prop
     *
     * @param error - the error which occurred
     * @param code - the error code to identify what error occurred
     * @param contextInfo - additional information which may be useful for the consumer of the error
     * @param origin - the origin of the error
     * @return void
     */
    handleError = (
        error: ElementsXhrError | Error,
        code: string,
        contextInfo: Record<string, unknown> = {},
        origin: ElementOrigin = this.props.errorOrigin,
    ): void => {
        if (!error || !code || !origin) {
            return;
        }

        const elementsError: ElementsError = {
            type: 'error',
            code,
            message: error.message,
            origin,
            context_info: {
                [IS_ERROR_DISPLAYED]: true,
                ...contextInfo,
            },
        };

        this.props.onError(elementsError);
    };

    render() {
        const { children, errorComponent: ErrorComponent, ...rest } = this.props;
        const { error } = this.state;
        if (error) {
            return <ErrorComponent error={error} />;
        }

        return React.cloneElement(children, {
            ...rest,
            onError: this.handleError,
        });
    }
}

export default ErrorBoundary;
