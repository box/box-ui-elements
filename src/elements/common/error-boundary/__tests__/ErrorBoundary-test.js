import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from '../ErrorBoundary';
import { ERROR_CODE_UNEXPECTED_EXCEPTION } from '../../../../constants';

describe('elements/ErrorBoundary', () => {
    const WrappedComponent = () => <div>Test</div>;
    const wrappedError = new Error('ERROR');

    const getWrapper = props =>
        shallow(
            <ErrorBoundary {...props}>
                <WrappedComponent />
            </ErrorBoundary>,
        );
    const simulateError = wrapper => {
        wrapper.find(WrappedComponent).simulateError(wrappedError);
    };

    describe('render()', () => {
        test('should render the wrapped component when no error is thrown', () => {
            const wrapper = getWrapper();

            expect(wrapper).toMatchSnapshot();
        });

        test('should render default error component an error is thrown', () => {
            const wrapper = getWrapper();

            simulateError(wrapper);

            expect(wrapper.find('DefaultError').exists()).toBe(true);
        });

        test('should render the component specified when an error is thrown', () => {
            const ErrorComponent = () => <div>Error</div>;
            const wrapper = getWrapper({
                errorComponent: ErrorComponent,
            });

            simulateError(wrapper);

            expect(wrapper.find(ErrorComponent).exists()).toBe(true);
        });
    });

    describe('componentDidCatch()', () => {
        const origin = 'some_component';
        test('should set the error state and call the onError callback', () => {
            const onError = jest.fn();
            const wrapper = getWrapper({
                onError,
                errorOrigin: origin,
            });

            simulateError(wrapper);

            expect(onError).toHaveBeenCalledWith({
                type: 'error',
                code: ERROR_CODE_UNEXPECTED_EXCEPTION,
                message: wrappedError.message,
                origin,
                context_info: expect.objectContaining({
                    isErrorDisplayed: true,
                }),
            });
        });
    });
});
