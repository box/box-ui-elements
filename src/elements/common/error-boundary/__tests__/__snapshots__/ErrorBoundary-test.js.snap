// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/ErrorBoundary render() should render the wrapped component when no error is thrown 1`] = `
<WrappedComponent
  onError={[Function]}
/>
`;
